File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {int x[2];int n=0;vector<int> c;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==target && n==0){
                x[0]=i;
                n++;
            }
            else if(nums[i]==target && n!=0){
                n++;
            }
        }
        x[1]=x[0]+n-1;
        if(n==0){
            x[0]=-1;
            x[1]=-1;
        }
        c.push_back(x[0]);
        c.push_back(x[1]);
        return c;
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] ans = {-1,-1};
        ans[0] = binary(nums,target,true);
        ans[1] = binary(nums,target,false);
        return ans;
    }
    static int binary(int[] nums,int target,boolean startind){
        int start = 0,end = nums.length-1;
        int ans = -1;
        while(start<=end){
            int mid = start + (end-start)/2;
            
            if(nums[mid] == target){
                ans = mid;
                if(startind){
                    end = mid - 1;
                }
                else{
                    start = mid  + 1;
                }
            }
            else if(nums[mid]>target) end = mid - 1;
            else start = mid + 1;
        }
        return ans;
    }
}